### DISCUSSION TITLE
0.6.2 is throwing an error

### DISCUSSION THREAD

**Discussion by BigZJD** [krzyzanowskim/STTextView#18] [Questions]
Error:
In STTextView+TextInputClient.swift

Type 'STTextView' does not conform to protocol 'NSTextInputClient'

Protocol requires function 'selectedRange()' with type '() -> NSRange' (aka '() -> _NSRange')

--------------------------------
I see that this should fix that:

     @objc(selectedRange)
         @_implements(NSTextInputClient, selectedRange())
         public func textInputClient_selectedRange() -> NSRange {
             selectedRange
         }
-------------------------------

Any ideas?

**Comment by luah5** [comment]
having the same issue, hoping that this gets resolved.

**Comment by krzyzanowskim** [comment]
> I see that this should fix that:

not sure what do you mean. This is how it is [already](https://github.com/krzyzanowskim/STTextView/blob/1380f59ad25d78d0d848a3b8de6bcbf3545b4b34/Sources/STTextView/STTextView%2BTextInputClient.swift#L77-L81C1) 

```swift
    @objc(selectedRange)
    @_implements(NSTextInputClient, selectedRange())
    public func textInputClient_selectedRange() -> NSRange {
        selectedRange
    }
```

**Reply by BigZJD** [reply]
Sorry, I should have been clearer. I meant that the above code should have taken care of the issue but wasn't.

**Comment by krzyzanowskim** [comment]
I think it's the compiler issue more than anything at this point. It does compile for a second build try.
I did move things around to please the compiler, the change is already pushed to the main branch

**Reply by BigZJD** [reply]
I appreciate the quick response. It works now and compiles. Thanks!

**Comment by swiftyfinch** [comment]
@krzyzanowskim Hello! I have the same issue but only if I'm trying to archive or build with the release configuration.
Could you help me?

I tried the latest version `0.6.3`, but it still builds correctly only with the debug configuration.
Xcode: 14.1 (14B47b)

<img width="371" alt="Screenshot 2023-05-14 at 16 30 37" src="https://github.com/krzyzanowskim/STTextView/assets/64660122/d5080b07-8ce3-41f6-97e2-a5db81cfd767">

<img width="760" alt="Screenshot 2023-05-14 at 16 30 50" src="https://github.com/krzyzanowskim/STTextView/assets/64660122/4d20bfd7-0941-4eaa-a0fb-9b8440f557bb">

**Reply by krzyzanowskim** [reply]
that sucks. I may need to change API to please the compiler then.

**Reply by swiftyfinch** [reply]
Yeah( But thank you for your quick response!

**Comment by krzyzanowskim** [comment]
I reverted most of the changes that cause the compiler to misbehave, that result in change to the API. Fix is pushed in https://github.com/krzyzanowskim/STTextView/releases/tag/0.6.4

**Reply by swiftyfinch** [reply]
Thanks a lot!

**Comment by luah5** [comment]


**Reply by krzyzanowskim** [reply]
that's on yours. you should call `setSelectedTextRange` instead
